tests/test_datacite.py

```diff
@@ -1,22 +1,119 @@
-from transmogrifier.helpers import parse_xml_records
-from transmogrifier.sources.datacite import Datacite
-
+from pytest import raises
 
-# TODO: add tests for expected data issues:
-# missing required Datacite fields raises error
-# all fields transformed successfully
-# repeatable fields transformed successfully
+from transmogrifier.models import Contributor, Date, Identifier, Note, TimdexRecord
+from transmogrifier.sources.datacite import Datacite
 
 
-def test_jpal_fixtures():
-    input_records = parse_xml_records("tests/fixtures/datacite_jpal_records.xml")
+def test_jpal_fixtures(datacite_jpal_records):
     output_records = Datacite(
         "jpal",
         "https://dataverse.harvard.edu/dataset.xhtml?persistentId=",
         "Abdul Latif Jameel Poverty Action Lab Dataverse",
-        input_records,
+        datacite_jpal_records,
     )
     assert len(list(output_records)) == 38
 
 
-# TODO: add tests for generate_name_identifier_url expected cases
+def test_datacite_full_record(datacite_record, datacite_jpal_record_full):
+    datacite_record.input_records = datacite_jpal_record_full
+    output_records = datacite_record
+    for output_record in output_records:
```

Comment:
Since we only expect a single output record, this will be more clear if we just use `assert next(output_records) == ...` (same for all tests below).